### DISCUSSION TITLE
Db connection usage of quarkus vs spring boot.

### DISCUSSION THREAD

**Discussion by samarism** [quarkusio/quarkus#44979] [Q&A]
Hello,

Is it true that quarkus releases a db connection into the pool as soon as it sends the query over the socket? I created a slow db query test for quarkus vs spring boot to understand this behavior. And saw that, quarkus released it's connection into the pool without waiting for the response while spring boot seemed to release it only after receiving the response from db.

[a comment by an automated account is not shown]

**Reply by cescoffier** [reply]
Nope—the connection stays open until the query terminates, even more so if you use transactions.

**Reply by geoand** [reply]
cc @yrodiere

**Reply by yrodiere** [reply]
> Nope—the connection stays open until the query terminates, even more so if you use transactions.

This. I don't know where you saw that Quarkus released a connection before getting a response, because I don't see how that could work. I mean I could dream of ways this could work, but not with the protocols used by today's relational databases.

**Reply by emmanuelbernard** [reply]
One thing Hibernate does is that it releases the connection right after the query is done. Which is not the case for every framework. But in case of a transactional event,t he poiol is most likely to keep the same connection for the next query.

**Reply by yrodiere** [reply]
> One thing Hibernate does is that it releases the connection right after the query is done. Which is not the case for every framework. But in case of a transactional event,t he poiol is most likely to keep the same connection for the next query.

Not sure about Hibernate in general or in WildFly/EAP/Spring, but in Quarkus we do this:

https://github.com/quarkusio/quarkus/blob/b4b93ada873b74250112ec0f70362b17af59c3df/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/boot/FastBootMetadataBuilder.java#L296-L307

So we release the connection just before the transaction completion... but even then, we release the *logical* connection, the one Agroal returns to us. Agroal will technically keep that connection "bound" to that transaction until the transaction is committed, and only then Agroal will return the connection to the pool.
